rustls/src/client/client_conn.rs

```diff
@@ -287,6 +287,11 @@ impl ClientConfig {
         }
     }
 
+    /// Return the crypto provider used to construct this client configuration.
+    pub fn crypto_provider(&self) -> Arc<CryptoProvider> {
```

Comment:
I think it would make sense to avoid the `Arc::clone()` inside the method, yielding `&Arc<_>` instead. This allows the caller to access the `CryptoProvider` without bumping the reference count. While bumping the reference count is cheap, it is not free, so I think we should allow the caller to decide whether they need it.

FWIW, I do still think `AsRef` is a nice fit for this. IMO `AsRef` (as opposed to `Deref`) conceptually implies more of a "has-a" relationship than an "is-a" relationship. But in the end this amounts to bikeshedding, so while I slightly prefer the `AsRef` solution I'm happy with either way.

Reply:
> I think it would make sense to avoid the Arc::clone() inside the method, yielding &Arc<_> instead. 

Makes sense to me :+1: Changed.

> while I slightly prefer the AsRef solution I'm happy with either way.

Since I think Ctz and myself both lightly prefer the accessor method approach I've gone with that instead of `AsRef`.